webapp/api_handlers.go

```diff
@@ -374,3 +377,122 @@ func handleAPIDiffPost(w http.ResponseWriter, r *http.Request) {
 	}
 	w.Write(bytes)
 }
+
+func apiManifestHandler(w http.ResponseWriter, r *http.Request) {
+	sha, err := ParseSHAParamFull(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ctx := appengine.NewContext(r)
+	if manifest, err := getManifestForSHA(ctx, sha); err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+	} else if manifest == nil {
+		http.NotFound(w, r)
+	} else {
+		w.Header().Add("content-type", "application/json")
+		w.Write(manifest)
+	}
+}
+
+func getManifestForSHA(ctx context.Context, sha string) (manifest []byte, err error) {
```

Comment:
Can you add some integration tests for this API? Do we have any tests that just hit the API endpoints, as opposed to invoking `getManifestForSHA` as we'd do here?

And... what do we do about GitHub API reliability in such tests? Mock out the GitHub API?

Reply:
Done, somewhat. IMHO abstracting to a `client` and asserting that we fetch a URL is as far as we should bother testing, otherwise you're writing tests that really only test how well you write your mocks.

Reply:
That makes sense, thanks!